sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/task/ExecutorServiceManager.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SonarQube Text Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.secrets.api.task;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provide an intermediately class to work with {@link ExecutorService}.
+ * It is dedicated to run tasks that are suspected to cause timeout, it will take care to stop them properly and ensure to keep a valid {@link ExecutorService}.
+ * The main entrypoint is the {@link #runWithTimeout} method.
+ */
+public class ExecutorServiceManager {
+  private static final Logger LOG = LoggerFactory.getLogger(ExecutorServiceManager.class);
+  /**
+   * The timeout time in millisecond after which the {@link ExecutorServiceManager} will try to interrupt the thread.
+   */
+  public static int timeoutMs = 10000;
+  /**
+   * The timeout time in millisecond after which the {@link ExecutorServiceManager} will stop waiting for the precedent interruption to be effective
+   * and will throw a {@link RuntimeException}
+   */
+  public static int uninterruptibleTimeoutMs = 10000;
```

Comment:
As we are effectively killing the analyzer after this time, we should definitely increase it. 
I would like 1000000 -> (16min) for this.